CustomCommand/ImportData/Console/Command/FileDataImporter.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace CustomCommand\ImportData\Console\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use CustomCommand\ImportData\Model\CustomerDataFactory;
+
+class FileDataImporter extends Command
+{
+    private const PROFILE = 'profile';
+
+    protected $dataFactory;
+    
+    public function __construct(CustomerDataFactory $dataFactory)
+    {
+        $this->dataFactory = $dataFactory;
+
+        parent::__construct();
+    }
+    protected function configure()
+    {
+        $options = [
+            new InputOption(
+                self::PROFILE,
+                '-s',
+                InputOption::VALUE_REQUIRED,
+                'Say the file extention'
+            ),
+        ];
+        $this->setName('customer:importer')
+            ->setDescription('Import customer data')
+            ->setDefinition($options)
+            ->addArgument('sourcePath', InputArgument::REQUIRED, 'Give source path');
+            parent::configure();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $source = (string)$input->getArgument('sourcePath');
+        $profile = $input->getOption('profile');
+        $output->writeln("Profile: $profile");
+        $output->writeln("Source: $source");
+        if ($sourceExtension = pathinfo($source, PATHINFO_EXTENSION)) {
+            switch ($sourceExtension) {
+                case 'json':
+                    $jsonContent = file_get_contents($source);
+                    $jsonData = json_decode($jsonContent, true);
+                    if ($jsonData === null) {
+                        $output->writeln("<error>Error parsing JSON file.</error>");
+                        return 1;
+                    }
+                    $output->writeln("<info>Data loaded successfully.</info>");
+                    $this->processJsonData($jsonData);
+                    return 0;
+                    break;
+                case 'csv':
+                    $csvData = array_map('str_getcsv', file($source));
+                    $csvHeaders = array_shift($csvData);
+                    if ($csvData === false) {
+                        $output->writeln("<error>Error reading CSV file.</error>");
+                        return 1;
+                    }
+                    $output->writeln("<info>Data loaded successfully.</info>");
+                    $this->processCsvData($csvData, $csvHeaders);
+                    return 0;
+                    break;
```

Comment:
You could have broken this into a smaller function definition and call the functions. Instead of writing the logic in the same method.